### ISSUE TITLE
[d3d9] Star Wars - Republic Commando crashes when starting a new game

### ISSUE THREAD

**Issue by WinterSnowfall** [doitsujin/dxvk#1430]
This does not happen with WineD3D, of course, hence the bug report. 

The intro videos have to be skipped because they get stuck while playing, but that's a Wine problem. Skipping them gets you on the main menu both with DXVK ...

![menu_dxvk](https://user-images.githubusercontent.com/6306593/74110151-c32c7300-4b92-11ea-9817-bc97991e3909.jpg)

... and WineD3D.

![menu_WineD3D](https://user-images.githubusercontent.com/6306593/74110152-c4f63680-4b92-11ea-93a0-0d2787cee4f8.jpg)

But starting a new game works only with WineD3D. 

![new_game_WineD3D](https://user-images.githubusercontent.com/6306593/74110216-49e15000-4b93-11ea-9eed-ef99f74bce81.jpg)

Being an Unreal Engine 2 title, I've used **d3d8to9** to patch it into DXVK and later into WineD3D, to confirm it works with the latter. To be honest, I was rather surprised when it crashed, and delayed reporting this until I was sure it wasn't something I did wrong, because all the other Unreal Engine 2 titles I've run this way so far performed admirably (Unreal Tournament 2004, Advent Rising).

To add insult to injury, I'm unable to get an apitrace with WineD3D, since the process makes the game crash right from the start. I was able, however, to get an apitrace with DXVK - I hope it will prove sufficient to shed some light on the matter.

### Software information
Star Wars - Republic Commando (GOG version)

### System information
- GPU: GeForce GTX 1070
- Driver: 440.59.0
- Wine version: 5.1 Staging
- DXVK version: v1.5.4

### Apitrace file(s)
[SWRepublicCommando_dxvk.trace.tar.xz](https://drive.google.com/open?id=19iZLE2NFmaQvRxd6GDUN2lbIJLpKA5Yd)

### Log files
- d3d9.log: [SWRepublicCommando_d3d9.log](https://github.com/doitsujin/dxvk/files/4177830/SWRepublicCommando_d3d9.log)
- d3d11.log: N/A
- dxgi.log: N/A

**Comment by WinterSnowfall** [comment]
So, 20 minutes later I had the great idea to set every graphical setting to rock bottom low just for the kicks and try again. Sure enough, starting a new game worked.

I narrowed it down to an in-game setting called: **Bumpmapping Quality**. If it's set to **Low**, things start working.

Here's an apitrace of the first few moments of a new game: [SWRepublicCommando_dxvk_low_bumpmap.trace.tar.xz](https://drive.google.com/open?id=1wHuAGVN4X3UE2mbndmKPRLznk8IvoyEv)

**Comment by K0bin** [comment]
> I narrowed it down to an in-game setting called: Bumpmapping Quality. If it's set to Low, things start working.

That's a game bug that also makes it crash on Windows. There's a mod that fixes it. https://www.moddb.com/mods/star-wars-republic-commando-graphics-fix/downloads/star-wars-republic-commando-graphics-fix

**Comment by WinterSnowfall** [comment]
> > I narrowed it down to an in-game setting called: Bumpmapping Quality. If it's set to Low, things start working.
> 
> That's a game bug that also makes it crash on Windows. There's a mod that fixes it. https://www.moddb.com/mods/star-wars-republic-commando-graphics-fix/downloads/star-wars-republic-commando-graphics-fix

Interesting... but it does work with WineD3D. I played it on Windows years ago and don't remember this issue, but that could just be a bug in my memory :). I'll try the fix with DXVK and report back, thanks for the link.

**Comment by Joshua-Ashton** [comment]
Bumpmapping on high on WineD3D works?

**Comment by WinterSnowfall** [comment]
> Bumpmapping on high on WineD3D works?

Weirdly, yes. The screenshot's above.

In any case, the mod fix shared by K0bin works like a charm with the added side-effect of chipping in support for 1080p/4K among other things, so I'm not one to complain.

You can mark this as a game-specific bug and close it.... unless you really want to investigate it further. Though if it's present in Windows as well, as per K0bin's feedback, I don't think it's worth the effort.

**Comment by Joshua-Ashton** [comment]
I'd be interested in fixing it, I think I have an idea what the problem is...

**Comment by K0bin** [comment]
I already discussed this on Discord with Josh but I'll just drop it here too for the sake of documentation.

The game tries to create a texture with the format L6V5U5, which DXVK (and most D3D9 drivers apparently) doesn't support.

`IDirect3DDevice9::CreateTexture(this = 0x839fdf8, Width = 512, Height = 512, Levels = 4, Usage = 0x0, Format = D3DFMT_L6V5U5, Pool = D3DPOOL_MANAGED, ppTexture = &0x85750d8, pSharedHandle = NULL) = D3D_OK`

While it should be possible to make that work, I think it's pretty low priority for now given that it doesnt work on Windows drivers either and that a workaround (the mod I posted earlier) exists.
